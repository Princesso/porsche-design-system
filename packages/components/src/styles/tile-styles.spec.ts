import { getTileStyles } from './tile-styles';
import type { JssStyle, Styles } from 'jss';

describe('getTileStyles()', () => {
  const additionalGlobalStyles: Styles = {
    p: {
      background: 'black',
    },
  };
  const additionalHostStyles: JssStyle = { cursor: 'not-allowed' };
  const additionalContentStyles: JssStyle = {
    gridTemplateRows: 'auto auto',
    gridTemplateColumns: 'auto',
  };

  it.each<Parameters<typeof getTileStyles>>([
    [{ aspectRatio: '4:3', isDisabled: true, additionalHostStyles, additionalGlobalStyles, additionalContentStyles }],
    [{ aspectRatio: '1:1', isDisabled: false, additionalHostStyles, additionalGlobalStyles, additionalContentStyles }],
    [{ aspectRatio: '3:4', isDisabled: true, additionalHostStyles, additionalGlobalStyles, additionalContentStyles }],
    [{ aspectRatio: '16:9', isDisabled: false, additionalHostStyles, additionalGlobalStyles, additionalContentStyles }],
    [{ aspectRatio: '9:16', isDisabled: true, additionalHostStyles, additionalGlobalStyles, additionalContentStyles }],
    [
      {
        aspectRatio: { base: '1:1', xs: '4:3', s: '3:4', m: '16:9', l: '9:16', xl: '1:1' },
        isDisabled: false,
        additionalHostStyles,
        additionalGlobalStyles,
        additionalContentStyles,
      },
    ],
  ])('should return correct css for %o', (args) => {
    expect(getTileStyles(args)).toMatchSnapshot();
  });
});
