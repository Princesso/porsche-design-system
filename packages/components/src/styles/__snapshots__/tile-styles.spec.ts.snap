// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTileStyles() should return correct css for aspectRatio: {"aspectRatio":"1:1","isDisabled":false,"additionalHostStyles":{"cursor":"not-allowed"},"additionalGlobalStyles":{"p":{"background":"black"}},"additionalContentStyles":{"gridTemplateRows":"auto auto","gridTemplateColumns":"auto"}}, isDisabled: %s additionalHostStyles: %j, additionalGlobalStyles: %j and additionalContentStyles: %j 1`] = `
{
  "@global": {
    "::slotted(img)": {
      "height": "100% !important",
      "objectFit": "cover !important",
      "width": "100% !important",
    },
    "::slotted(picture)": {
      "bottom": "0 !important",
      "left": "0 !important",
      "position": "absolute !important",
      "right": "0 !important",
      "top": "0 !important",
    },
    "::slotted(picture),::slotted(img)": {
      "WebkitBackfaceVisibility": "hidden !important",
      "backfaceVisibility": "hidden !important",
      "transition": "transform var(--p-transition-duration, .24s) ease !important",
    },
    ":host": {
      "&([hidden])": {
        "display": "none !important",
      },
      "cursor": "not-allowed !important",
      "display": "block",
    },
    "p": {
      "background": "black",
    },
  },
  "content": {
    "@media (forced-colors: active)": {
      "background": "rgba(0,0,0,0.7)",
    },
    "borderRadius": "8px",
    "gap": "16px",
    "gridTemplateColumns": "auto",
    "gridTemplateRows": "auto auto",
    "justifyItems": "start",
    "left": 0,
    "position": "absolute",
    "right": 0,
  },
  "image-container": {
    "borderRadius": "8px",
    "bottom": 0,
    "left": 0,
    "overflow": "hidden",
    "position": "absolute",
    "right": 0,
    "top": 0,
    "transform": "translateZ(0)",
  },
  "root": {
    "@media(hover:hover)": {
      "&:hover": {
        "& ::slotted(picture),::slotted(img)": {
          "transform": "scale3d(1.05, 1.05, 1.05) !important",
        },
      },
    },
    "height": 0,
    "paddingTop": "100%",
    "position": "relative",
    "transform": "translate3d(0,0,0)",
  },
}
`;

exports[`getTileStyles() should return correct css for aspectRatio: {"aspectRatio":"3:4","isDisabled":true,"additionalHostStyles":{"cursor":"not-allowed"},"additionalGlobalStyles":{"p":{"background":"black"}},"additionalContentStyles":{"gridTemplateRows":"auto auto","gridTemplateColumns":"auto"}}, isDisabled: %s additionalHostStyles: %j, additionalGlobalStyles: %j and additionalContentStyles: %j 1`] = `
{
  "@global": {
    "::slotted(img)": {
      "height": "100% !important",
      "objectFit": "cover !important",
      "width": "100% !important",
    },
    "::slotted(picture)": {
      "bottom": "0 !important",
      "left": "0 !important",
      "position": "absolute !important",
      "right": "0 !important",
      "top": "0 !important",
    },
    "::slotted(picture),::slotted(img)": {
      "WebkitBackfaceVisibility": "hidden !important",
      "backfaceVisibility": "hidden !important",
      "transition": "transform var(--p-transition-duration, .24s) ease !important",
    },
    ":host": {
      "&([hidden])": {
        "display": "none !important",
      },
      "cursor": "not-allowed !important",
      "display": "block",
    },
    "p": {
      "background": "black",
    },
  },
  "content": {
    "@media (forced-colors: active)": {
      "background": "rgba(0,0,0,0.7)",
    },
    "borderRadius": "8px",
    "gap": "16px",
    "gridTemplateColumns": "auto",
    "gridTemplateRows": "auto auto",
    "justifyItems": "start",
    "left": 0,
    "position": "absolute",
    "right": 0,
  },
  "image-container": {
    "borderRadius": "8px",
    "bottom": 0,
    "left": 0,
    "overflow": "hidden",
    "position": "absolute",
    "right": 0,
    "top": 0,
    "transform": "translateZ(0)",
  },
  "root": {
    "height": 0,
    "paddingTop": "133.33%",
    "position": "relative",
    "transform": "translate3d(0,0,0)",
  },
}
`;

exports[`getTileStyles() should return correct css for aspectRatio: {"aspectRatio":"4:3","isDisabled":true,"additionalHostStyles":{"cursor":"not-allowed"},"additionalGlobalStyles":{"p":{"background":"black"}},"additionalContentStyles":{"gridTemplateRows":"auto auto","gridTemplateColumns":"auto"}}, isDisabled: %s additionalHostStyles: %j, additionalGlobalStyles: %j and additionalContentStyles: %j 1`] = `
{
  "@global": {
    "::slotted(img)": {
      "height": "100% !important",
      "objectFit": "cover !important",
      "width": "100% !important",
    },
    "::slotted(picture)": {
      "bottom": "0 !important",
      "left": "0 !important",
      "position": "absolute !important",
      "right": "0 !important",
      "top": "0 !important",
    },
    "::slotted(picture),::slotted(img)": {
      "WebkitBackfaceVisibility": "hidden !important",
      "backfaceVisibility": "hidden !important",
      "transition": "transform var(--p-transition-duration, .24s) ease !important",
    },
    ":host": {
      "&([hidden])": {
        "display": "none !important",
      },
      "cursor": "not-allowed !important",
      "display": "block",
    },
    "p": {
      "background": "black",
    },
  },
  "content": {
    "@media (forced-colors: active)": {
      "background": "rgba(0,0,0,0.7)",
    },
    "borderRadius": "8px",
    "gap": "16px",
    "gridTemplateColumns": "auto",
    "gridTemplateRows": "auto auto",
    "justifyItems": "start",
    "left": 0,
    "position": "absolute",
    "right": 0,
  },
  "image-container": {
    "borderRadius": "8px",
    "bottom": 0,
    "left": 0,
    "overflow": "hidden",
    "position": "absolute",
    "right": 0,
    "top": 0,
    "transform": "translateZ(0)",
  },
  "root": {
    "height": 0,
    "paddingTop": "75%",
    "position": "relative",
    "transform": "translate3d(0,0,0)",
  },
}
`;

exports[`getTileStyles() should return correct css for aspectRatio: {"aspectRatio":"9:16","isDisabled":true,"additionalHostStyles":{"cursor":"not-allowed"},"additionalGlobalStyles":{"p":{"background":"black"}},"additionalContentStyles":{"gridTemplateRows":"auto auto","gridTemplateColumns":"auto"}}, isDisabled: %s additionalHostStyles: %j, additionalGlobalStyles: %j and additionalContentStyles: %j 1`] = `
{
  "@global": {
    "::slotted(img)": {
      "height": "100% !important",
      "objectFit": "cover !important",
      "width": "100% !important",
    },
    "::slotted(picture)": {
      "bottom": "0 !important",
      "left": "0 !important",
      "position": "absolute !important",
      "right": "0 !important",
      "top": "0 !important",
    },
    "::slotted(picture),::slotted(img)": {
      "WebkitBackfaceVisibility": "hidden !important",
      "backfaceVisibility": "hidden !important",
      "transition": "transform var(--p-transition-duration, .24s) ease !important",
    },
    ":host": {
      "&([hidden])": {
        "display": "none !important",
      },
      "cursor": "not-allowed !important",
      "display": "block",
    },
    "p": {
      "background": "black",
    },
  },
  "content": {
    "@media (forced-colors: active)": {
      "background": "rgba(0,0,0,0.7)",
    },
    "borderRadius": "8px",
    "gap": "16px",
    "gridTemplateColumns": "auto",
    "gridTemplateRows": "auto auto",
    "justifyItems": "start",
    "left": 0,
    "position": "absolute",
    "right": 0,
  },
  "image-container": {
    "borderRadius": "8px",
    "bottom": 0,
    "left": 0,
    "overflow": "hidden",
    "position": "absolute",
    "right": 0,
    "top": 0,
    "transform": "translateZ(0)",
  },
  "root": {
    "height": 0,
    "paddingTop": "177.75%",
    "position": "relative",
    "transform": "translate3d(0,0,0)",
  },
}
`;

exports[`getTileStyles() should return correct css for aspectRatio: {"aspectRatio":"16:9","isDisabled":false,"additionalHostStyles":{"cursor":"not-allowed"},"additionalGlobalStyles":{"p":{"background":"black"}},"additionalContentStyles":{"gridTemplateRows":"auto auto","gridTemplateColumns":"auto"}}, isDisabled: %s additionalHostStyles: %j, additionalGlobalStyles: %j and additionalContentStyles: %j 1`] = `
{
  "@global": {
    "::slotted(img)": {
      "height": "100% !important",
      "objectFit": "cover !important",
      "width": "100% !important",
    },
    "::slotted(picture)": {
      "bottom": "0 !important",
      "left": "0 !important",
      "position": "absolute !important",
      "right": "0 !important",
      "top": "0 !important",
    },
    "::slotted(picture),::slotted(img)": {
      "WebkitBackfaceVisibility": "hidden !important",
      "backfaceVisibility": "hidden !important",
      "transition": "transform var(--p-transition-duration, .24s) ease !important",
    },
    ":host": {
      "&([hidden])": {
        "display": "none !important",
      },
      "cursor": "not-allowed !important",
      "display": "block",
    },
    "p": {
      "background": "black",
    },
  },
  "content": {
    "@media (forced-colors: active)": {
      "background": "rgba(0,0,0,0.7)",
    },
    "borderRadius": "8px",
    "gap": "16px",
    "gridTemplateColumns": "auto",
    "gridTemplateRows": "auto auto",
    "justifyItems": "start",
    "left": 0,
    "position": "absolute",
    "right": 0,
  },
  "image-container": {
    "borderRadius": "8px",
    "bottom": 0,
    "left": 0,
    "overflow": "hidden",
    "position": "absolute",
    "right": 0,
    "top": 0,
    "transform": "translateZ(0)",
  },
  "root": {
    "@media(hover:hover)": {
      "&:hover": {
        "& ::slotted(picture),::slotted(img)": {
          "transform": "scale3d(1.05, 1.05, 1.05) !important",
        },
      },
    },
    "height": 0,
    "paddingTop": "56.25%",
    "position": "relative",
    "transform": "translate3d(0,0,0)",
  },
}
`;

exports[`getTileStyles() should return correct css for aspectRatio: {"aspectRatio":{"base":"1:1","xs":"4:3","s":"3:4","m":"16:9","l":"9:16","xl":"1:1"},"isDisabled":false,"additionalHostStyles":{"cursor":"not-allowed"},"additionalGlobalStyles":{"p":{"background":"black"}},"additionalContentStyles":{"gridTemplateRows":"auto auto","gridTemplateColumns":"auto"}}, isDisabled: %s additionalHostStyles: %j, additionalGlobalStyles: %j and additionalContentStyles: %j 1`] = `
{
  "@global": {
    "::slotted(img)": {
      "height": "100% !important",
      "objectFit": "cover !important",
      "width": "100% !important",
    },
    "::slotted(picture)": {
      "bottom": "0 !important",
      "left": "0 !important",
      "position": "absolute !important",
      "right": "0 !important",
      "top": "0 !important",
    },
    "::slotted(picture),::slotted(img)": {
      "WebkitBackfaceVisibility": "hidden !important",
      "backfaceVisibility": "hidden !important",
      "transition": "transform var(--p-transition-duration, .24s) ease !important",
    },
    ":host": {
      "&([hidden])": {
        "display": "none !important",
      },
      "cursor": "not-allowed !important",
      "display": "block",
    },
    "p": {
      "background": "black",
    },
  },
  "content": {
    "@media (forced-colors: active)": {
      "background": "rgba(0,0,0,0.7)",
    },
    "borderRadius": "8px",
    "gap": "16px",
    "gridTemplateColumns": "auto",
    "gridTemplateRows": "auto auto",
    "justifyItems": "start",
    "left": 0,
    "position": "absolute",
    "right": 0,
  },
  "image-container": {
    "borderRadius": "8px",
    "bottom": 0,
    "left": 0,
    "overflow": "hidden",
    "position": "absolute",
    "right": 0,
    "top": 0,
    "transform": "translateZ(0)",
  },
  "root": {
    "@media(hover:hover)": {
      "&:hover": {
        "& ::slotted(picture),::slotted(img)": {
          "transform": "scale3d(1.05, 1.05, 1.05) !important",
        },
      },
    },
    "@media(min-width:1000px)": {
      "paddingTop": "56.25%",
    },
    "@media(min-width:1300px)": {
      "paddingTop": "177.75%",
    },
    "@media(min-width:1760px)": {
      "paddingTop": "100%",
    },
    "@media(min-width:480px)": {
      "paddingTop": "75%",
    },
    "@media(min-width:760px)": {
      "paddingTop": "133.33%",
    },
    "height": 0,
    "paddingTop": "100%",
    "position": "relative",
    "transform": "translate3d(0,0,0)",
  },
}
`;
